src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiOptionsFormFields.js

```diff
@@ -0,0 +1,360 @@
+import {produce} from 'immer';
+import React, {useContext, useMemo, useState} from 'react';
+import {useIntl} from 'react-intl';
+import useAsync from 'react-use/esm/useAsync';
+
+import Loader from 'components/admin/Loader';
+import {CustomFieldTemplate} from 'components/admin/RJSFWrapper';
+import {Checkbox, NumberInput, TextArea, TextInput} from 'components/admin/forms/Inputs';
+import Select from 'components/admin/forms/Select';
+import {ValidationErrorContext} from 'components/admin/forms/ValidationErrors';
+import {get} from 'utils/fetch';
+
+import {
+  REGISTRATION_OBJECTTYPES_ENDPOINT,
+  REGISTRATION_OBJECTTYPE_VERSIONS_ENDPOINT,
+} from '../../constants';
+import {getErrorMarkup, getFieldErrors} from './utils';
+
+const Wrapper = ({children}) => (
+  <form className="rjsf" name="form.registrationBackendOptions">
+    <CustomFieldTemplate displayLabel={false} errors={null}>
+      <fieldset id="root">{children}</fieldset>
+    </CustomFieldTemplate>
+  </form>
+);
+
+const ObjectsApiOptionsFormFields = ({index, name, schema, formData, onChange}) => {
+  const intl = useIntl();
+  const validationErrors = useContext(ValidationErrorContext);
+
+  const [availableObjecttypes, setAvailableObjecttypes] = useState([]);
+  const [availableObjecttypeVersions, setAvailableObjecttypeVersions] = useState([]);
+
+  const {
+    objecttype = '',
+    objecttypeVersion = '',
+    productaanvraagType = '',
+    informatieobjecttypeSubmissionReport = '',
+    uploadSubmissionCsv = '',
+    informatieobjecttypeSubmissionCsv = '',
+    informatieobjecttypeAttachment = '',
+    organisatieRsin = '',
+    contentJson = '',
+    paymentStatusUpdateJson = '',
+  } = formData;
+
+  // For existing registration options where the URL might not be valid
+  // will default to a textinput in that case
+  const unknownObjecttype = useMemo(
+    () => objecttype && !availableObjecttypes.map(o => o.url).includes(objecttype),
+    [availableObjecttypes, objecttype]
+  );
+
+  const onFieldChange = event => {
+    const {name, value} = event.target;
+    const updatedFormData = produce(formData, draft => {
+      draft[name] = value;
+    });
+    onChange(updatedFormData);
+  };
+
+  const buildErrorsComponent = field => {
+    const rawErrors = getFieldErrors(name, index, validationErrors, field);
+    return rawErrors ? getErrorMarkup(rawErrors) : null;
+  };
+
+  // API fetches:
+  const {loading: objecttypesLoading} = useAsync(async () => {
+    const response = await get(REGISTRATION_OBJECTTYPES_ENDPOINT);
+    if (!response.ok) {
+      throw new Error('Error when loading objecttypes');
+    }
```

Comment:
I think that these errors are being swallowed somewhere :thinking:

Reply:
Can confirm by raising in the backend. Probably related to how `useAsync` or hooks work? I don't really know how these errors should be handled. Another example in the codebase, where this time it is ignored:

https://github.com/open-formulieren/open-forms/blob/537cddd1f83953409608ec91bcca2745d4125424/src/openforms/js/components/admin/form_design/variables/VariablesTable.js#L130-L139

Reply:
`useAsync` returns `{loading, value, error}` so exception should show up in the `error` key.

you may also need explicit `try ... catch` and re-throw the error because of how `async/await` work in JS :unamused:

Reply:
I grabbed the error key to also fallback to textinputs, as if no objecttype client is configured, the available objecttypes can't be retrieved and we need to provide the legacy way